app/views/links/show.html.erb

```diff
@@ -1,5 +1,42 @@
 <% content_for :meta do %>
   <link href="<%= @product.long_url %>" rel="canonical">
+
+  <% if @product.native_type == Link::NATIVE_TYPE_EBOOK %>
+    <%
```

Comment:
Please move this JSON construction logic to a dedicated instance method of the `Link` model, ideally defined in a concern that the `Link` model would include.